src/OpenLoco/src/GameCommands/VehicleOrderDelete.cpp

```diff
@@ -0,0 +1,53 @@
+#include "Entities/EntityManager.h"
+#include "GameCommands/GameCommands.h"
+#include "Ui/WindowManager.h"
+#include "Vehicles/OrderManager.h"
+#include "Vehicles/Orders.h"
+#include "Vehicles/Vehicle.h"
+#include <OpenLoco/Interop/Interop.hpp>
+
+using namespace OpenLoco::Interop;
+
+namespace OpenLoco::GameCommands
+{
+    // 0x0047057A
+    static uint32_t vehicleOrderDelete(const VehicleOrderDeleteArgs& args, uint8_t flags)
+    {
+        auto* head = EntityManager::get<Vehicles::VehicleHead>(args.head);
+        if (head == nullptr)
+        {
+            return GameCommands::FAILURE;
+        }
+
+        GameCommands::setPosition(head->position);
+        if (!(flags & GameCommands::Flags::apply))
+        {
+            return 0;
+        }
+
+        Ui::WindowManager::sub_4B93A5(enumValue(head->id));
+
+        // Find out what type the selected order is
```

Comment:
Appear to be missing the part that moves the current order if its past the deletion point.

Reply:
Wait, I remember implementing that?? I'll look into it, thanks.

Reply:
You were right, it was missing. Since implemented.